npm/private/npm_import.bzl

```diff
@@ -276,59 +278,91 @@ def npm_link_imported_package(
+    target = "%s:%s/%s/%s" % (link_workspace_and_package, utils.package_store_root, link_root_name, package_store_name)
     _directory_path(
         name = "%s__entry_point" % name,
-        directory = "@{link_workspace}//{root_package}:{{}}/dir".format(store_target_name),
-        path = "{bin_path}",
+        directory = target + "/dir",
+        path = bin_path,
         tags = ["manual"],
     )
     _js_binary(
         name = "%s__js_binary" % name,
         entry_point = ":%s__entry_point" % name,
-        data = ["@{link_workspace}//{root_package}:{{}}".format(store_target_name)],
+        data = [target],
         include_npm = kwargs.pop("include_npm", False),
         tags = ["manual"],
     )
     _js_run_binary(
         name = name,
         tool = ":%s__js_binary" % name,
-        mnemonic = kwargs.pop("mnemonic", "{bin_mnemonic}"),
+        mnemonic = kwargs.pop("mnemonic", bin_mnemonic),
         **kwargs
     )
 
-def _{bin_name}_test_internal(name, link_root_name, **kwargs):
-    store_target_name = "{package_store_root}/{{}}/{package_store_name}".format(link_root_name)
+def bin_test_internal(name, link_workspace_and_package, link_root_name, package_store_name, bin_path, **kwargs):
+    target = "%s:%s/%s/%s" % (link_workspace_and_package, utils.package_store_root, link_root_name, package_store_name)
     _directory_path(
         name = "%s__entry_point" % name,
-        directory = "@{link_workspace}//{root_package}:{{}}/dir".format(store_target_name),
-        path = "{bin_path}",
+        directory = target + "/dir",
+        path = bin_path,
         tags = ["manual"],
     )
     _js_test(
         name = name,
         entry_point = ":%s__entry_point" % name,
-        data = kwargs.pop("data", []) + ["@{link_workspace}//{root_package}:{{}}".format(store_target_name)],
+        data = kwargs.pop("data", []) + [target],
         **kwargs
     )
 
-def _{bin_name}_binary_internal(name, link_root_name, **kwargs):
-    store_target_name = "{package_store_root}/{{}}/{package_store_name}".format(link_root_name)
+def bin_binary_internal(name, link_workspace_and_package, link_root_name, package_store_name, bin_path, **kwargs):
+    target = "%s:%s/%s/%s" % (link_workspace_and_package, utils.package_store_root, link_root_name, package_store_name)
     _directory_path(
         name = "%s__entry_point" % name,
-        directory = "@{link_workspace}//{root_package}:{{}}/dir".format(store_target_name),
-        path = "{bin_path}",
+        directory = target + "/dir",
+        path = bin_path,
         tags = ["manual"],
     )
     _js_binary(
         name = name,
         entry_point = ":%s__entry_point" % name,
-        data = kwargs.pop("data", []) + ["@{link_workspace}//{root_package}:{{}}".format(store_target_name)],
+        data = kwargs.pop("data", []) + [target],
         **kwargs
     )
 
+_BIN_MACRO_TMPL = """
```

Comment:
I think this macro can change to just have `_{bin_name}_path = "{bin_path}"` so the `"{bin_path}"` doesn't get repeated 3x?